content/en/blog/_posts/2021-03-18-introducing-suspended-jobs.md

```diff
@@ -0,0 +1,100 @@
+solve use-cases that require Pods to run forever, Jobs are useful when Pods need
+to run to completion. Commonly used in parallel batch processing, Jobs can be
+used in a variety of applications ranging from video rendering and database
+maintenance to sending bulk emails and scientific computing.
+
+While the amount of parallelism and the conditions for Job completion are
+configurable, the Kubernetes API lacked the ability to suspend and resume Jobs.
+This is often desired when cluster resources are limited and a higher priority
+Job needs to execute in the place of another Job. Deleting the lower priority
+Job is a poor workaround as Pod completion history and other metrics associated
+with the Job will be lost.
+
+With the recent Kubernetes 1.21 release, you will be able to suspend a Job by
+updating its Job spec. The feature is currently in **alpha** and requires you to
+enable the `SuspendJob` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+on the [API server](/docs/reference/command-line-tools-reference/kube-apiserver/)
+and the [controller manager](/docs/reference/command-line-tools-reference/kube-controller-manager/)
+in order to use it.
+
+## API changes
+
+A new boolean field `suspend` is introduced in the Job spec API. Let's say I
+create the following Job.
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: my-job
+spec:
+  suspend: true
+  parallelism: 2
+  completions: 10
+  template:
+    spec:
+      containers:
+      - name: my-container
+        image: busybox
+        command: ["sleep", "5"]
+      restartPolicy: Never
+```
+
+Jobs are not suspended by default, so I need to explicitly set the `suspend`
+field to true in the above Job spec. The Job controller will now refrain from
+creating Pods until I'm ready to start the Job, which I can do by updating the
+field to false.
+
+If I update this field from false back to true again, the Job controller will
+terminate all active Pods it created. This is done by sending the Pods a SIGTERM
+signal; the [graceful termination period](/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination)
+defined in the Pod spec will be honoured and Pods terminated this way will not
+be counted as failures by the Job controller.
+
+Read the [documentation](/docs/concepts/workloads/controllers/job#suspending-a-job)
+for a full overview of the implications of using this new feature.
+
+## Where is this useful?
+
+Let's say I'm the operator of a large cluster. I have many users submitting Jobs
+to the cluster, but not all Jobs are created equal — some Jobs are more
+important than others. Cluster resources aren't infinite either, so all users
+must share resources. If all Jobs were created in the suspended state and placed
+in a pending queue, I can achieve priority-based Job scheduling by resuming Jobs
+in the right order.
+
+As another motivational use-case, consider a cloud provider where compute
```

Comment:
we need to be careful with the why this use case is described because we count completions from scratch every time the job is unsuspended. I would suggest not to include this use case or clearly callout that the application needs to have the ability to understand how to continue after a suspend.

Reply:
finalized pods are not removed (unless they are GCed). But pod progress is lost.

Reply:
Because these shortcomings could be surprising, I think we _should_ mention a case where they might be relevant, and then explain the problem. Maybe mention Indexed completion as a way for Pods to work out what progress to (separately) track.

Reply:
>we count completions from scratch every time the job is unsuspended

Perhaps I've misunderstood, but we do NOT count from scratch. Successfully completed pods in the past will continue to count towards the completions counter even if the Job is suspended and resumed. Pods that were *removed* as a direct result of the suspension won't be counted, of course, but I thought that is fairly obvious from the note about SIGTERM in the previous section.

Reply:
Not directly obvious, I left a comment on the paragraph above.